cci_cameroon/analysis/text_similarity/similarity_search_sample.py

```diff
@@ -0,0 +1,608 @@
+if torch.cuda.is_available():
+    model2 = model2.to(torch.device("cuda"))
+print(model2.device)
+
+# %%
+# produce sentence vectors
+embeddings = model2.encode(sentences)
+
+# %%
+# Step 1: Change data type
+embeddings2 = np.array([embedding for embedding in embeddings]).astype("float32")
+
+# %%
+print(embeddings[:10])
+
+# %%
+print(embeddings2[:10])
+
+# %%
+# create a faiss index for the search using the embenddings
+index_dis = faiss.IndexFlatL2(embeddings.shape[1])
+
+# %%
+index_dis.add(embeddings)  # add the vector to the index created
+
+# %%
+# search the index for our given input xq
+# %time
+distance4, position4 = index_dis.search(xq, k)  # search
+# get the text
+for i in position4[0]:
+    print(sentences[i])
+print(distance4[0])
+
+# %%
+sentences[position4[0][0]]
+
+# %%
+num_cells = 50  # Number of cells the index should be divided into
+quantizer = faiss.IndexFlatL2(dimension)
+index_dis2 = faiss.IndexIVFFlat(quantizer, dimension, num_cells)
+
+# %%
+index_dis2.train(embeddings)
+index_dis2.add(embeddings)
+
+# %%
+# search the index for our given input xq
+# %time
+distance5, position5 = index_dis2.search(xq, k)  # search
+# get the text
+for i in position5[0]:
+    print(sentences[i])
+print(distance5[0])
+
+# %%
+# print("Similarity:", util.dot_score(xq, embeddings)[0][5168])
+# We use cosine-similarity and torch.topk to find the highest 5 scores
+# cos_scores = util.pytorch_cos_sim(xq, embeddings)[0]
+# top_results = torch.topk(cos_scores, k=k)
+
+# %% [markdown]
+# # Using French models
+# We create a sample corpus of French sentences that would be used across the different pre-trained models below.
+
+# %%
+sentences_french = [
+    "C'est une personne heureuse",
+    "J'aime mon téléphone",
+    "Mon téléphone n'est pas bon.",
+    "Elle est heureuse apres avoir eu le BAC",
+    "Votre téléphone portable est superbe.",
+    "Récemment, de nombreux ouragans ont frappé les États-Unis",
+    "Aujourd'hui est une journée ensoleillée",
+]
+sample_french1 = ["J'aime le téléphone que j'ai!"]
+sample_french2 = ["C'est une personne très heureuse"]
+
+# %%
+print(torch.__version__)
+
+# %% [markdown]
+# ## 1. https://www.pinecone.io/learn/faiss-tutorial/
+# 2. Choosing index https://towardsdatascience.com/billion-scale-semantic-similarity-search-with-faiss-sbert-c845614962e2
+# 3. Points to consider when choosing an index: https://github.com/facebookresearch/faiss/wiki/Guidelines-to-choose-an-index
+# 4. distilBert impl mis-info covid  https://github.com/kstathou/vector_engine/blob/master/notebooks/001_vector_search.ipynb
+#
+# 5. Roberta https://pytorch.org/hub/pytorch_fairseq_roberta/
+#
+# 6. Camembert is French implementation of Roberta https://github.com/pytorch/fairseq/tree/main/examples/camembert
+# 7. Paper with multilingual text embeddings - frence : https://jep-taln2020.loria.fr/wp-content/uploads/JEP-TALN-RECITAL-2020_paper_209.pdf MUSE and M-BERT are chosen.
+#
+#
+#
+
+# %%
+# #!pip install bitarray
```

Comment:
I know this is more because of install issues but for the final cleaned notebook I would have any imports or instructions on what to install at the top.